sdv/metadata/single_table.py

```diff
@@ -245,11 +244,92 @@ def to_dict(self):
 
         return deepcopy(metadata)
 
+    def _determine_sdtype_for_numbers(self, data):
+        """Determine the sdtype for a numerical column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = 'numerical'
+        if len(data) > 5:
+
+            whole_values = (data == data.round()).loc[~data.isna()].all()
+            positive_values = (data >= 0).loc[~data.isna()].all()
+
+            unique_values = data.nunique()
+            percentage_unique_values = unique_values <= round(len(data) / 10)
+
+            if whole_values and positive_values and percentage_unique_values:
+                sdtype = 'categorical'
+            elif unique_values == len(data) and whole_values:
+                sdtype = 'id'
+
+        return sdtype
+
+    def _determine_sdtype_for_objects(self, data):
+        """Determine the sdtype for an object column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = None
+        try:
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=UserWarning)
```

Comment:
What warnings are being shown here and why we ignore them? 🤔

Reply:
It was because we didn't give any `datetime_format` and we ignored it because the user couldn't do anything about it. But now I removed it after the changes of [4aa4a4a](https://github.com/sdv-dev/SDV/pull/1529/commits/4aa4a4ac2c96acb7a1fa5929cc5826512e128fc8)